tests/unit-tests/03-number/_num.scm

```diff
@@ -777,36 +777,36 @@
 (check-eqv? (* 1+i 1.) 1.+1.i)
 (check-eqv? (* 1+i 1+i) +2i)
 
-(let ((arg 1)) (check-eq? arg (* 1 arg)))
-(let ((arg 1)) (check-eq? arg (* arg 1)))
-(let ((arg 1)) (check-eq? arg (+ 0 arg)))
-(let ((arg 1)) (check-eq? arg (+ arg 0)))
-(let ((arg 1)) (check-eq? arg (/ arg 1)))
-(let ((arg 1)) (check-eq? arg (- arg 0)))
-(let ((arg 1267650600228229401496703205376)) (check-eq? arg (* 1 arg)))
-(let ((arg 1267650600228229401496703205376)) (check-eq? arg (* arg 1)))
-(let ((arg 1267650600228229401496703205376)) (check-eq? arg (+ 0 arg)))
-(let ((arg 1267650600228229401496703205376)) (check-eq? arg (+ arg 0)))
-(let ((arg 1267650600228229401496703205376)) (check-eq? arg (/ arg 1)))
-(let ((arg 1267650600228229401496703205376)) (check-eq? arg (- arg 0)))
-(let ((arg 1/2)) (check-eq? arg (* 1 arg)))
-(let ((arg 1/2)) (check-eq? arg (* arg 1)))
-(let ((arg 1/2)) (check-eq? arg (+ 0 arg)))
-(let ((arg 1/2)) (check-eq? arg (+ arg 0)))
-(let ((arg 1/2)) (check-eq? arg (/ arg 1)))
-(let ((arg 1/2)) (check-eq? arg (- arg 0)))
-(let ((arg 1.)) (check-eq? arg (* 1 arg)))
-(let ((arg 1.)) (check-eq? arg (* arg 1)))
-(let ((arg 1.)) (check-eq? arg (+ 0 arg)))
-(let ((arg 1.)) (check-eq? arg (+ arg 0)))
-(let ((arg 1.)) (check-eq? arg (/ arg 1)))
-(let ((arg 1.)) (check-eq? arg (- arg 0)))
-(let ((arg 1+i)) (check-eq? arg (* 1 arg)))
-(let ((arg 1+i)) (check-eq? arg (* arg 1)))
-(let ((arg 1+i)) (check-eq? arg (+ 0 arg)))
-(let ((arg 1+i)) (check-eq? arg (+ arg 0)))
-(let ((arg 1+i)) (check-eq? arg (/ arg 1)))
-(let ((arg 1+i)) (check-eq? arg (- arg 0)))
+(let ((arg 1)) (check-equal? arg (* 1 arg)))
```

Comment:
The code in `_num.scm` is written to preserve object identity in these cases, that's why I used `check-eq?`.  Is there a reason I don't know why your changing them?

Reply:
There's no reason to require of these operations to preserve object identity.  What is required is that the value and exactness of arg is preserved.  When gsc compiles this code the expansion of these operations would have to be overly complex to preserve object identity.

Reply:
It appears that if you add
```
(declare (not inline-primitives))
```
to the top of
```
tests/unit-tests/#.scm
```
then all the original tests in `_num.scm` using `check-eq?` pass with
```
firefly:~/programs/gambit/gambit/tests> ./run-unit-tests.scm -C -gsc
```
I really think of the tests in `unit-tests` as runtime library tests,  not compiler tests, and I think we should make this change to `#.scm` (and put the old tests back, too).

Reply:
I have nothing against the idea of testing the library, and the `(declare (not inline-primitives))` is fine for that.  But even for the library the preservation of object identity should not be a requirement of the test.  I want the test to represent what we expect from the library procedure.  I want to be able to change the implementation of the library and not have to change the unit tests.  So only the properties that are required should be tested and nothing else, otherwise we would end up having tests that depend on a specific implementation that is overly constraining.

Independently, the exact same tests should be run with `(declare (inline-primitives))` (as is the case now) so that the compiler's inlining can be tested too.  This is true for all tests, so maybe the `run-unit-tests.scm` script should have an option to select which declaration should be used.

BTW, these double tests are a good motivation for testing what is expected of the procedures (so that they work whether they end up being inlined or an actual call to the library).